src/withStyles.jsx

```diff
@@ -67,7 +70,23 @@ export function withStyles(
             }}
           />
         );
-      }
+      };
+    } else {
+      // WrappedComponent is a function, so we can avoid a little overhead by
+      // calling it directly instead of rendering it. This avoids React
+      // lifecycle overhead for the wrapped component and may not be necessary
+      // in a future version of React.
+      WithStyles.prototype.render = function render() {
+        const { themeName } = this.context;
+        flushIfNecessary();
+        return WrappedComponent({
```

Comment:
Yes, this is true but we can fix it when that happens.

I believe most of the overhead is likely in react lifecycle methods, but I haven't taken the time to dig into specifics. As I mentioned in my commit message, I expect that this will be optimized by a future version of React, but in the meantime we can get better performance now.

Not entirely sure about propType checks. Can you give me an example? One option might be to only apply this optimization in production.